ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/time/DelayMechanism.scala

```diff
@@ -6,20 +6,25 @@ package com.daml.ledger.api.testtool.infrastructure.time
 import com.daml.ledger.api.testtool.infrastructure.participant.ParticipantTestContext
 import com.daml.timer.Delayed
 
+import scala.concurrent.duration.Duration
 import scala.concurrent.{ExecutionContext, Future}
-import scala.concurrent.duration.{Duration, FiniteDuration}
 
 trait DelayMechanism {
-  val skews: FiniteDuration
   def delayBy(duration: Duration): Future[Unit]
 }
 
-class TimeDelayMechanism(val skews: FiniteDuration)(implicit ec: ExecutionContext)
-    extends DelayMechanism {
+object DelayMechanism {
+  def apply(ledger: ParticipantTestContext)(implicit ec: ExecutionContext): DelayMechanism =
+    if (ledger.features.staticTime) {
+      new StaticTimeDelayMechanism(ledger)
+    } else new TimeDelayMechanism()
```

Comment:
This could potentially be a method on `ParticipantTestContext`, so someone could use `ledger.delayMechanism`, or even `ledger.delayBy` directly. What do you think?

Reply:
Good point, I'll move the delayMechanism as part of the `ParticipantTestContext`